app/services/rate_creator.rb

```diff
@@ -13,17 +14,17 @@ def initialize(value, submission, user, rate_checker)
   end
 
   def call
-    rate = Rate.new({ value: @value, submission: @submission, user: @user })
-
-    if @rate_checker.user_has_already_rated?(@submission.id, @user.id)
-        errors = ['You are not allowed to rate the same submission twice']
-        success = false
-        result = Result.new(rate, success, errors)
+    if rate_checker.user_has_already_rated?
```

Comment:
Now that I think about it... Do we need this `user_has_already_rated?` and the conditional at all?
Couldn't we do it in these three lines:
```Ruby
def call
  rate = Rate.find_or_create(user_id: user.id, submission_id: submission.id)
  success = rate.update(value: value)
  Result.new(rate, success)
end
```